### PR TITLE
Add runtime timeout

### PR DESCRIPTION
**Pull request by JaafarRammal**
Add timeout in case of bugs like deadlocks or infinite loops to not kill the pipeline for 6 hours

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -61,7 +61,8 @@ jobs:
     
   run:
     name: Run
-    runs-on: ubuntu-latest 
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
```
Comment (jzzheng22) [inline .github/workflows/main.yaml:65]:
```suggestion
    timeout-minutes: 15
```

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yaml (hunk 1 above)
Comment (jzzheng22) [reply .github/workflows/main.yaml:65]: Actually idk if this is necessary

Comment (JaafarRammal) [reply .github/workflows/main.yaml:65]: The timeout or the 15?

Comment (jzzheng22) [reply .github/workflows/main.yaml:65]: The 15

Comment (JaafarRammal) [reply .github/workflows/main.yaml:65]: Let's put 10 hehe

**Review by jzzheng22 — APPROVED** [review]

**Pull request merged**